### ISSUE TITLE
Custom created directory hashes can forge sizes for files, affecting "CumulativeSize" from object stat

### ISSUE THREAD

**Issue by nannal** [ipfs/go-ipfs#4151]
go-ipfs version: 0.4.10-
Repo version: 5
System version: amd64/linux
Golang version: go1.8.3

Bug: Low Priority 

Using dag put it's possible the write a directory object 

{"Links":[{"Name":"08-2017.txt","Hash":"QmaLgbbyusb4mmUTYzhoHyapE1HvTYeTrmDkEToC5NGHmc","Size":666666666666666}],"Data":"\u0008\u0001"}

The object size can be any int value 

This affects the cumulative size of the object

ipfs object stat QmXrfRCyBoyh5T8nnEA589GcmPgeGQz1mKm5RPaKB6QutT
NumLinks: 1
BlockSize: 64
LinksSize: 62
DataSize: 2
CumulativeSize: 666666666666730

While this isn't critical functionality it's something I personally use frequently for displaying the size of a file.

My alternative is to run object links and tally up the BlockSize to create a total but this requires all the children's content be available.

**Comment by timthelion** [comment]
How would you envision this behavior being different? It seems to me that it is either necesary to look at the data, or have accurate metadata. And if your metadata is inaccurate, then what can be done but to look at the data?

**Comment by nannal** [comment]
I agree it's necessary to either look at the data or have accurate metadata. 

Since we cant assure accurate metadata we must have the client calculate the value, no peers should be able to provide inaccurate metadata to the network as the information provided is presented as fact not speculation.

**Comment by Stebalien** [comment]
Unfortunately, that would make for very poor UX. It's quite useful to know the expected size of a file before one attempts to download it.

However, IPFS could reject files with incorrect sizes as malformed/invalid.

**Comment by whyrusleeping** [comment]
Yeah, I think the best solution here is to add some validators to ipfs that throw errors if expectations arent lining up

**Comment by Kubuxu** [comment]
Also there is problem with overflow in those structures. I've seen huge directories overflow the size.
